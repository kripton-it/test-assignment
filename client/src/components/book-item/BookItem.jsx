import React from 'react';

import * as Styles from './styles';

import { trim, maxSynopsisLength } from '../../utils';

class BookItem extends React.Component {
  render() {
    const { title, author, cover, synopsis, slug } = this.props.book;
    const url = `/book/${slug}`;
    const trimmedSynopsis = trim(synopsis, maxSynopsisLength);
    return (
      <Styles.BookItemContainer>
        <Styles.BookItemAuthor>{author}</Styles.BookItemAuthor>
        <Styles.BookLink to={url}>
          <Styles.BookItemTitle>{title}</Styles.BookItemTitle>
        </Styles.BookLink>

        <Styles.BookItemSynopsis>{trimmedSynopsis}</Styles.BookItemSynopsis>
        <Styles.BookItemCover>
          <Styles.BookItemImg src={cover} alt={title} />
        </Styles.BookItemCover>
      </Styles.BookItemContainer>
    );
  }
}

export default BookItem;