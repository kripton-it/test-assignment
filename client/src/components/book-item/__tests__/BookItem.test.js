import React from "react";
import { shallow } from "enzyme";
import toJson from 'enzyme-to-json';

import BookItem from "../BookItem";

const book = {
  title: 'title',
  author: 'author',
  cover: 'cover',
  synopsis: 'synopsis',
  slug: 'slug'
};

it("expect to render BookItem component", () => {
  const wrapper = shallow(<BookItem book={book} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
