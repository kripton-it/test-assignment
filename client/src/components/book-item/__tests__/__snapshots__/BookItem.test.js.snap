// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render BookItem component 1`] = `
<BookItemContainer>
  <BookItemCover>
    <BookCoverLink
      to="/book/slug"
    >
      <BookItemImg
        alt="title"
        src="cover"
      />
    </BookCoverLink>
  </BookItemCover>
  <BookTitleLink
    to="/book/slug"
  >
    <BookItemTitle>
      title
    </BookItemTitle>
  </BookTitleLink>
  <BookItemAuthor>
    author
  </BookItemAuthor>
  <BookItemSynopsis>
    synopsis
  </BookItemSynopsis>
</BookItemContainer>
`;
